solrmonitor/solrmonitor.go

```diff
@@ -18,23 +18,32 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"strconv"
 	"strings"
 	"sync"
 	"sync/atomic"
 
 	"github.com/fullstorydev/zk"
 )
 
+const (
+	collectionsPath    = "/collections"
+	liveNodesPath      = "/live_nodes"
+	liveQueryNodesPath = "/live_query_nodes"
+)
+
 // Keeps an in-memory copy of the current state of the Solr cluster; automatically updates on ZK changes.
 type SolrMonitor struct {
 	logger    zk.Logger // where to debug log
 	zkCli     ZkCli     // the ZK client
 	solrRoot  string    // e.g. "/solr"
 	zkWatcher *ZkWatcherMan
 
-	mu          sync.RWMutex
-	collections map[string]*collection // map of all currently-known collections
-	liveNodes   []string               // current set of live_nodes
+	mu                sync.RWMutex
+	collections       map[string]*collection // map of all currently-known collections
+	liveNodes         []string               // current set of live_nodes
+	queryNodes        []string               // current set of live_query_nodes
```

Comment:
^^ could we eliminate some of these state variables entirely, in favor of a pure event publishing model?  It seems beneficial to have the client track state entirely.

Reply:
I think livenodes, querynodes we can remove. Currently, solrman is referring to that, but probably I can take it next time.